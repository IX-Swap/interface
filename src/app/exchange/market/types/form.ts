export interface PlaceOrderFormValues {
  type: string
  timeInForce: string
  price: number
  amount: number
  slider: number
  total: number
}

export interface PlaceOrderArgs {
  pair: string
  side: string
  type: string
  price: number
  amount: number
}
