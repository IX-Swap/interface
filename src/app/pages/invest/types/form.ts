export interface PlaceOrderFormValues {
  price: number
  amount: number
  total: number
}

export interface PlaceOrderArgs {
  pair: string
  side: string
  type: string
  price: number
  amount: number
}
