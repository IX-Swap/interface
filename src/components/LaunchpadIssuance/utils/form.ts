export const isSubmitDisabled = (errors: any, touched: any) => {
  return getError(errors, touched)
}

const getIsDisabled = (message?: string) => {
  if (message) {
    const includesString = (keys: string[]) => keys.some((key: string) => message.includes(key))
    const invalidKeys = ['valid', 'Valid']
    const requiredKeys = ['required', 'Required', 'enter', 'Enter', 'Add at least']
    return includesString(invalidKeys) || !includesString(requiredKeys)
  }
  return false
}

const getError = (errors: any, touched: any) => {
  const hasError = Object.entries(touched).some(([fieldName, value]: any) => {
    if (Array.isArray(value)) {
      return value.some((arrayItem, index) => {
        if (!arrayItem) {
          return false
        }
        return Object.keys(arrayItem).some((arrayItemKey) => {
          const error = errors[fieldName] && errors[fieldName][index] && errors[fieldName][index][arrayItemKey]
          return getIsDisabled(error)
        })
      })
    }
    if (typeof value === 'object') {
      return Object.keys(value).some((objectItemKey) => {
        const error = errors[fieldName] && errors[fieldName][objectItemKey]
        return getIsDisabled(error)
      })
    }

    return value ? getIsDisabled(errors[fieldName]) : false
  })
  return hasError
}

export const isDraftDisabled = (errors: any, touched: any) => {
  return getError(errors, touched)
}
