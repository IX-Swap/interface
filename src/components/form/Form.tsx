import React, { PropsWithChildren } from 'react'
import { SubmitHandler, useForm, FormProvider } from 'react-hook-form'
import { ObjectSchema, Shape, object } from 'yup'
import { yupResolver } from '@hookform/resolvers'
import { useUnmountCallback } from 'hooks/useUnmountCallback'
import { useServices } from 'hooks/useServices'

export interface FormProps<T extends {}> {
  defaultValues?: Partial<T>
  onSubmit?: SubmitHandler<T>
  validationSchema?: ObjectSchema<Shape<object | undefined, T>>
  criteriaMode?: 'all' | 'firstError'
}

export const Form = <T,>(props: PropsWithChildren<FormProps<T>>) => {
  const {
    defaultValues,
    onSubmit = console.log,
    validationSchema = object({}),
    criteriaMode = 'firstError',
    children,
    ...rest
  } = props

  const form = useForm({
    mode: 'onBlur',
    defaultValues: defaultValues as any,
    resolver: yupResolver(validationSchema),
    criteriaMode: criteriaMode
  })
  const { snackbarService } = useServices()
  const onError = () => {
    void snackbarService.showSnackbar(
      'Please fill all required fields',
      'error'
    )
  }

  useUnmountCallback(() => {
    if (form.formState.isDirty) {
      form.reset()
    }
  })

  return (
    <FormProvider {...form}>
      <form
        {...rest}
        style={{ width: '100%' }}
        onSubmit={form.handleSubmit(onSubmit, onError)}
      >
        {children}
      </form>
      {/* <pre style={{ fontSize: 14 }}>
        {JSON.stringify(form.getValues(), null, 4)}
      </pre> */}
    </FormProvider>
  )
}
