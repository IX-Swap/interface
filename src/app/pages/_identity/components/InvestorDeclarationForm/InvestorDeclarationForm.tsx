import React, { useEffect } from 'react'
import { Grid, Typography } from '@material-ui/core'
import { FormSectionHeader } from 'app/pages/_identity/components/FormSectionHeader'
import { DeclarationsListFields } from 'app/pages/_identity/components/InvestorDeclarationForm/DeclarationsList/DeclartionsListFields'
import { IdentityType } from 'app/pages/identity/utils'
import { OptInAgreements } from 'app/pages/_identity/components/InvestorDeclarationForm/OptInAgreements/OptInAgreements'
import { InvestorAgreements } from 'app/pages/_identity/components/InvestorDeclarationForm/InvestorAgreements/InvestorAgreements'
import { useFormContext } from 'react-hook-form'
import { useServices } from 'hooks/useServices'

export interface InvestorDeclarationFormProps {
  identityType?: IdentityType
}

export const InvestorDeclarationForm = ({
  identityType = 'individual'
}: InvestorDeclarationFormProps) => {
  const { snackbarService } = useServices()
  const { errors } = useFormContext()
  const declarationsError = errors.investorDeclarations

  useEffect(() => {
    if (declarationsError !== undefined) {
      void snackbarService.showSnackbar(declarationsError.message, 'error')
    }
  }, [declarationsError]) // eslint-disable-line

  return (
    <>
      <FormSectionHeader title='Investor Status Declaration' />
      <Grid container spacing={3}>
        <Grid item xs={12}>
          <Typography>
            Singapore rules require you to declare your investor status before
            you see live deals on our platform
          </Typography>
        </Grid>
        <Grid item xs={12}>
          <InvestorAgreements type={identityType} />
        </Grid>

        <Grid item xs={12}>
          <FormSectionHeader title={'Opt-In Requirement'} />
          <DeclarationsListFields
            title='I confirm to be treated as an “Accredited Investor” by InvestaX'
            data={[
              {
                name: 'optInAgreements',
                label: <OptInAgreements showOptOutDialog />
              }
            ]}
          />
        </Grid>
      </Grid>
    </>
  )
}
