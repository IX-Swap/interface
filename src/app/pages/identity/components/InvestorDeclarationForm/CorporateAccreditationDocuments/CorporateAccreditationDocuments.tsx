import React from 'react'
import {
  Grid,
  List,
  ListItem as MuiListItem,
  ListItemIcon,
  Typography
} from '@mui/material'
import { UploadDocumentField } from 'app/pages/identity/components/UploadDocumentsForm/UploadDocumentField/UploadDocumentField'
import { FieldContainer } from 'ui/FieldContainer/FieldContainer'
import { InvestorRole } from 'app/pages/identity/utils/shared'
import { FormSectionHeader } from 'ui/FormSectionHeader/FormSectionHeader'

export interface ListItemProps {
  children: React.ReactNode
}

export const ListItem = ({ children }: ListItemProps) => (
  <MuiListItem style={{ paddingLeft: 6, alignItems: 'flex-start' }}>
    <ListItemIcon style={{ minWidth: 16 }}>&#8226;</ListItemIcon>
    {children}
  </MuiListItem>
)

export interface CorporateAccreditationDocumentsProps {
  corporateType?: 'investor' | 'issuer'
  investorRole?: InvestorRole
}

export const CorporateAccreditationDocuments = (
  props: CorporateAccreditationDocumentsProps
) => {
  const { corporateType = 'issuer', investorRole = 'accredited' } = props
  const isIssuer = corporateType === 'issuer'
  const isExpert = investorRole === 'expert'

  return (
    <Grid item>
      <FieldContainer>
        <Grid container direction={'column'} spacing={5}>
          <Grid item>
            <FormSectionHeader title={'Accreditation Documents'} />
          </Grid>
          {/* <Grid item>
            <UploadDocumentField
              name='financialDocuments'
              label='Financial Documents'
              helperElement={
                <Typography color={'text.secondary'} mt={1.5} fontWeight={400}>
                  Balance sheet, profit & loss statement or annual returns
                </Typography>
              }
            />
          </Grid> */}
          {!isIssuer && (
            <Grid item>
              <UploadDocumentField
                name='evidenceOfAccreditation'
                label={`Evidence of ${
                  !isExpert ? 'Accreditation' : 'Expertise'
                }`}
                helperElement={
                  <Grid item container direction='column'>
                    {!isExpert ? (
                      <Grid item>
                        <List>
                          <ListItem>
                            <Typography
                              color={'text.secondary'}
                              fontWeight={400}
                            >
                              Copy of the most recent audited balance sheet of
                              the corporation.
                            </Typography>
                          </ListItem>
                          <ListItem>
                            <Typography
                              color={'text.secondary'}
                              fontWeight={400}
                            >
                              Where the corporation is not required to prepare
                              audited account regularly, a balance sheet of the
                              corporation of the state of affairs of the
                              corporation as of the date of the balance sheet,
                              of which date shall be within the preceding 12
                              months.
                            </Typography>
                          </ListItem>
                        </List>
                      </Grid>
                    ) : (
                      <Grid item>
                        <Typography
                          color={'text.secondary'}
                          fontWeight={400}
                          my={1}
                        >
                          For Example: SGX trading member confirmation, recent
                          custody or broker trading statement, confirmation
                          letter from broker/custodian for trading activity for
                          past 12 months, financial statement/reports showing
                          previous engagement or dealing in capital market
                          products, trust deed showing intended activities of
                          the trust.
                        </Typography>
                      </Grid>
                    )}
                    <Grid item>
                      <Typography color={'text.secondary'} fontWeight={400}>
                        *Note that the above list is not exhaustive.
                      </Typography>
                    </Grid>
                  </Grid>
                }
              />
            </Grid>
          )}

          {/* Hidden upload field for documents in Corporate KYC. This is to preserve their value in the `documents` field */}
          <Grid item hidden>
            <UploadDocumentField
              name='corporateDocuments'
              label='Corporate Documents'
            />
          </Grid>
        </Grid>
      </FieldContainer>
    </Grid>
  )
}
