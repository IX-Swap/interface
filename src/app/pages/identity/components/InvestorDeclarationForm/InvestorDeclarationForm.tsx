import React, { useEffect } from 'react'
import { Grid, Typography } from '@mui/material'
import { FormSectionHeader } from 'app/pages/identity/components/FormSectionHeader'
import { DeclarationsListFields } from 'app/pages/identity/components/InvestorDeclarationForm/DeclarationsList/DeclartionsListFields'
import { OptInAgreements } from 'app/pages/identity/components/InvestorDeclarationForm/OptInAgreements/OptInAgreements'
import { InvestorAgreements } from 'app/pages/identity/components/InvestorDeclarationForm/InvestorAgreements/InvestorAgreements'
import { useFormContext } from 'react-hook-form'
import { IdentityType } from 'app/pages/identity/utils/shared'
import { FieldContainer } from 'app/pages/identity/components/FieldContainer/FieldContainer'
import { CorporateDocuments } from 'app/pages/identity/components/InvestorDeclarationForm/CorporateDocuments/CorporateDocuments'
import { InstitutionalInvestorAgreements } from 'app/pages/identity/components/InvestorDeclarationForm/InstitutionalInvestorAgreements/InstitutionalInvestorAgreements'
import { Divider } from 'ui/Divider'
import { UploadDocumentField } from 'app/pages/identity/components/UploadDocumentsForm/UploadDocumentField/UploadDocumentField'

export interface InvestorDeclarationFormProps {
  identityType?: IdentityType
  corporateType?: 'investor' | 'issuer'
}

export const InvestorDeclarationForm = ({
  identityType = 'individual',
  corporateType = 'issuer'
}: InvestorDeclarationFormProps) => {
  const { formState, trigger } = useFormContext()

  const {
    assets,
    trustee,
    accreditedShareholders,
    partnership,
    accreditedBeneficiaries,
    accreditedSettlors,
    optInAgreements,
    digitalSecurities,
    primaryOfferingServices,
    digitalSecuritiesIssuance,
    allServices,
    isInstitutionalInvestor
  } = formState.dirtyFields

  useEffect(() => {
    void trigger()
  }, [
    assets,
    trustee,
    accreditedShareholders,
    partnership,
    accreditedBeneficiaries,
    accreditedSettlors,
    optInAgreements,
    digitalSecurities,
    primaryOfferingServices,
    digitalSecuritiesIssuance,
    allServices,
    isInstitutionalInvestor,
    trigger
  ])

  return (
    <Grid container direction={'column'} spacing={2}>
      <Grid
        item
        data-testid='investorStatusDeclaration'
        container
        spacing={2}
        direction={'column'}
      >
        <Grid item xs={12}>
          <FieldContainer>
            <Grid container spacing={3} direction={'column'}>
              <Grid item>
                <FormSectionHeader title='Investor Status Declaration' />
              </Grid>
              <Grid item>
                <InvestorAgreements type={identityType} />
              </Grid>
            </Grid>
          </FieldContainer>
        </Grid>

        <Grid item xs={12}>
          <FieldContainer>
            <Grid container direction={'column'} spacing={5}>
              <Grid item container spacing={3} direction={'column'}>
                <Grid item>
                  <FormSectionHeader title={'Opt-In Requirement'} />
                </Grid>
                <Grid item>
                  <DeclarationsListFields
                    title='I confirm to be treated as an “Accredited Investor” by InvestaX'
                    data={[
                      {
                        name: 'optInAgreements',
                        label: <OptInAgreements showOptOutDialog />
                      }
                    ]}
                  />
                </Grid>
              </Grid>
            </Grid>
          </FieldContainer>
        </Grid>
      </Grid>

      <Grid item xs={12}>
        <FieldContainer>
          <Grid item container direction={'column'}>
            <Grid item>
              <FormSectionHeader title='Institutional Investor Status Declaration' />
            </Grid>

            <Grid item>
              <DeclarationsListFields
                title=''
                data={[
                  {
                    name: 'isInstitutionalInvestor',
                    label: <InstitutionalInvestorAgreements />
                  }
                ]}
              />
            </Grid>

            <Grid item mt={3} mb={3}>
              <Divider />
            </Grid>

            <Grid item>
              <Grid item>
                <UploadDocumentField
                  name='institutionalInvestorDocuments'
                  label='Institutional Investor Documents'
                  helperElement={
                    <Typography
                      color={'text.secondary'}
                      mt={1.5}
                      fontWeight={400}
                    >
                      Institutional Investor Documents - License issued by
                      Regulator
                    </Typography>
                  }
                />
              </Grid>
            </Grid>
          </Grid>
        </FieldContainer>
      </Grid>

      {identityType !== 'individual' && (
        <CorporateDocuments corporateType={corporateType} />
      )}

      {/* <ValidateOnMount /> */}
    </Grid>
  )
}
