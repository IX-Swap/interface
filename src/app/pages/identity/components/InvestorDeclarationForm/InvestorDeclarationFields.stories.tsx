import React from 'react'
import { Meta, Story } from '@storybook/react/types-6-0'
import { InvestorDeclarationForm } from 'app/pages/identity/components/InvestorDeclarationForm/InvestorDeclarationForm'
import { Grid } from '@material-ui/core'
import { Form } from 'components/form/Form'

const meta: Meta = {
  title: 'Pages/Identity/InvestorDeclarationForm',
  component: InvestorDeclarationForm
}

export default meta

const Template: Story = () => {
  return (
    <Form>
      <Grid container spacing={6}>
        <Grid item xs={12}>
          <InvestorDeclarationForm />
        </Grid>
      </Grid>
    </Form>
  )
}
export const Default = Template.bind({})
