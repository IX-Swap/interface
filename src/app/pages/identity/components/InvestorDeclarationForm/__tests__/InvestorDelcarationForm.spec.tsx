import React from 'react'
import { render } from 'test-utils'
import { InvestorDeclarationForm } from 'app/pages/identity/components/InvestorDeclarationForm/InvestorDeclarationForm'
import { CorporateDocuments } from 'app/pages/identity/components/InvestorDeclarationForm/CorporateAccreditationDocuments/CorporateAccreditationDocuments'
import { Form } from 'components/form/Form'

jest.mock(
  'app/pages/identity/components/InvestorDeclarationForm/CorporateDocuments/CorporateDocuments',
  () => ({
    CorporateDocuments: jest.fn(() => null)
  })
)

describe('InvestorDeclarationForm', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = render(
      <Form>
        <InvestorDeclarationForm />
      </Form>
    )

    expect(container).toMatchSnapshot()
  })

  it('renders corporate documents when identityType is corporate', () => {
    render(
      <Form>
        <InvestorDeclarationForm identityType={'corporate'} />
      </Form>
    )
    expect(CorporateDocuments).toHaveBeenCalled()
  })
})
