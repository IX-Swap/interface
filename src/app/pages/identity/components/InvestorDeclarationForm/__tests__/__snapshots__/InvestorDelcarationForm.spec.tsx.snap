// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InvestorDeclarationForm should match snapshot 1`] = `
<div>
  <form
    style="width: 100%;"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-1rl1og4-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column css-10h8c1q-MuiGrid-root"
        data-testid="investorStatusDeclaration"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1r7sl14-MuiPaper-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-32q0xd-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiBox-root css-0"
                >
                  <h5
                    class="MuiTypography-root MuiTypography-h5 css-1br8tjs-MuiTypography-root"
                  >
                    Investor Status Declaration
                  </h5>
                </div>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <p
                  class="MuiTypography-root MuiTypography-body1 css-1b4l2pg-MuiTypography-root"
                >
                  I declare that I am "Individual Accredited Investor"
                </p>
              </div>
              <div
                class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      id="personalAssets"
                      path="personalAssets"
                      style="align-items: flex-start;"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="personalAssets"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                        style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                      >
                        My total net personal assets (including up to SGD 1 million of your primary residence) exceed SGD 2 million
                      </p>
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      id="income"
                      path="income"
                      style="align-items: flex-start;"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="income"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                        style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                      >
                        My income in the preceding 12 months is not less than SGD 300,000 (or its equivalent in a foreign currency)
                      </p>
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      id="financialAsset"
                      path="financialAsset"
                      style="align-items: flex-start;"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="financialAsset"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                        style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                      >
                        My personal financial asset (e.g. deposits and investment product) exceed SGD 1 million or its equivalent
                      </p>
                    </label>
                  </div>
                  <div
                    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                  >
                    <label
                      class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                      id="jointlyHeldAccount"
                      path="jointlyHeldAccount"
                      style="align-items: flex-start;"
                    >
                      <span
                        class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                      >
                        <input
                          class="PrivateSwitchBase-input css-1m9pwf3"
                          data-indeterminate="false"
                          name="jointlyHeldAccount"
                          type="checkbox"
                        />
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                          data-testid="CheckBoxOutlineBlankIcon"
                          focusable="false"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                          />
                        </svg>
                        <span
                          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                        />
                      </span>
                      <p
                        class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                        style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                      >
                        My jointly held account with my spouse/any individual meets any of the above
                      </p>
                    </label>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation0 css-1r7sl14-MuiPaper-root"
          >
            <div
              class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 MuiGrid-direction-xs-column css-vacfyj-MuiGrid-root"
            >
              <div
                class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-3 MuiGrid-direction-xs-column css-ttp6t5-MuiGrid-root"
              >
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiBox-root css-0"
                  >
                    <h5
                      class="MuiTypography-root MuiTypography-h5 css-1br8tjs-MuiTypography-root"
                    >
                      Opt-In Requirement
                    </h5>
                  </div>
                </div>
                <div
                  class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
                >
                  <div
                    class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
                  >
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1c439vl-MuiGrid-root"
                    >
                      <h6
                        class="MuiTypography-root MuiTypography-subtitle1 makeStyles-title-1 css-1b4l2pg-MuiTypography-root"
                        data-testid="title-typo"
                      >
                        I confirm to be treated as an “Accredited Investor” by InvestaX
                      </h6>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                    >
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        id="optInAgreements1"
                        path="optInAgreements1"
                        style="align-items: flex-start;"
                      >
                        <span
                          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            name="optInAgreements1"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                          style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                        >
                          I have been informed of and understand the consequences of my qualification as an Accredited Investor, in particular the reduced regulatory investor 
                          <span
                            class="MuiTypography-root MuiTypography-body1 css-movqfk-MuiTypography-root"
                            style="cursor: pointer;"
                          >
                            safeguards
                          </span>
                           for Accredited Investors.
                        </p>
                      </label>
                    </div>
                    <div
                      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-ewf807-MuiGrid-root"
                    >
                      <label
                        class="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-j204z7-MuiFormControlLabel-root"
                        id="optInAgreements2"
                        path="optInAgreements2"
                        style="align-items: flex-start;"
                      >
                        <span
                          class="MuiCheckbox-root MuiCheckbox-colorPrimary MuiButtonBase-root MuiCheckbox-root MuiCheckbox-colorPrimary PrivateSwitchBase-root css-o5n8uc-MuiButtonBase-root-MuiCheckbox-root"
                        >
                          <input
                            class="PrivateSwitchBase-input css-1m9pwf3"
                            data-indeterminate="false"
                            name="optInAgreements2"
                            type="checkbox"
                          />
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1oaawcc-MuiSvgIcon-root"
                            data-testid="CheckBoxOutlineBlankIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
                            />
                          </svg>
                          <span
                            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                          />
                        </span>
                        <p
                          class="MuiTypography-root MuiTypography-body1 css-govr2p-MuiTypography-root"
                          style="margin-top: 8px; font-size: 14px; line-height: 1.5;"
                        >
                          I have been informed of and understand my right to
                           
                          <span
                            class="MuiTypography-root MuiTypography-body1 css-movqfk-MuiTypography-root"
                            style="cursor: pointer;"
                          >
                            opt out
                          </span>
                           of the Accredited Investors status
                        </p>
                      </label>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </form>
</div>
`;
